tests/integration/kubernetes/gha-run.sh

```diff
@@ -157,83 +158,64 @@ function deploy_kata() {
+		HOST_OS=${KATA_HOST_OS}
 	fi
 
 	if [ "${KATA_HYPERVISOR}" = "qemu" ]; then
-		yq -i \
-		  '.spec.template.spec.containers[0].env[6].value = "image initrd kernel default_vcpus"' \
-		  "${tools_dir}/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml"
+		ALLOWED_HYPERVISOR_ANNOTATIONS="image initrd kernel default_vcpus"
 	fi
 
 	if [ "${KATA_HYPERVISOR}" = "qemu-tdx" ]; then
-		yq -i \
-		  ".spec.template.spec.containers[0].env[8].value = \"${HTTPS_PROXY}\"" \
-		  "${tools_dir}/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml"
-
-		yq -i \
-		  ".spec.template.spec.containers[0].env[9].value = \"${NO_PROXY}\"" \
-		  "${tools_dir}/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml"
+		AGENT_HTTPS_PROXY="${HTTPS_PROXY}"
+		AGENT_NO_PROXY="${NO_PROXY}"
 	fi
 
 	# Set the PULL_TYPE_MAPPING
 	if [ "${PULL_TYPE}" != "default" ]; then
-		yq -i \
-		  ".spec.template.spec.containers[0].env[10].value = \"${KATA_HYPERVISOR}:${PULL_TYPE}\"" \
-		  "${tools_dir}/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml"
+		PULL_TYPE_MAPPING="${KATA_HYPERVISOR}:${PULL_TYPE}"
 	fi
 
+	local values_yaml
+	values_yaml=$(mktemp /tmp/values_yaml.XXXXXX)
+
+	cat <<-EOF > "${values_yaml}"
+	k8sDistribution: ${KUBERNETES}
+	image:
+	  reference: ${DOCKER_REGISTRY}/${DOCKER_REPO}
+	  tag: ${DOCKER_TAG}
+	env:
+	  debug: "true"
+	  shims: "${KATA_HYPERVISOR}"
+	  defaultShim: "${KATA_HYPERVISOR}"
+	  createRuntimeClass: "true"
+	  createDefaultRuntimeClass: "true"
+	  allowedHypervisorAnnotations: "${ALLOWED_HYPERVISOR_ANNOTATIONS}"
+	  snapshotterHandlerMapping: "${SNAPSHOTTER_HANDLER_MAPPING}"
+	  agentHttpsProxy: "${AGENT_HTTPS_PROXY}"
+	  agentNoProxy: "${AGENT_NO_PROXY}"
+	  pullTypeMapping: "${PULL_TYPE_MAPPING}"
+	  hostOS: "${HOST_OS}"
+	EOF
+
 	echo "::group::Final kata-deploy.yaml that is used in the test"
-	cat "${tools_dir}/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml"
-	grep "${DOCKER_REGISTRY}/${DOCKER_REPO}:${DOCKER_TAG}" "${tools_dir}/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml" || die "Failed to setup the tests image"
+	helm template ./"${helm_chart_dir}" --set-file values.yaml="${values_yaml}" --namespace kube-system
+	sed -n "/reference: ${DOCKER_REGISTRY}\/${DOCKER_REPO}/{N;/\n[[:space:]]*tag: ${DOCKER_TAG}/p;}" "${values_yaml}"
 	echo "::endgroup::"
 
-	kubectl_retry apply -f "${tools_dir}/packaging/kata-deploy/kata-rbac/base/kata-rbac.yaml"
-	case "${KUBERNETES}" in
-		k0s) kubectl_retry apply -k "${tools_dir}/packaging/kata-deploy/kata-deploy/overlays/k0s" ;;
-		k3s) kubectl_retry apply -k "${tools_dir}/packaging/kata-deploy/kata-deploy/overlays/k3s" ;;
-		rke2) kubectl_retry apply -k "${tools_dir}/packaging/kata-deploy/kata-deploy/overlays/rke2" ;;
-		*) kubectl_retry apply -f "${tools_dir}/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml"
-	esac
+	helm install kata-deploy ./"${helm_chart_dir}" --set-file values.yaml="${values_yaml}" --namespace kube-system
```

Comment:
@beraldoleal Here is "your" install :) using this method we also apply any helm chart changes to the CI